homeassistant/components/intellidrive/cover.py

```diff
@@ -0,0 +1,153 @@
+"""The cover entity of reisinger intellidrive."""
+
+import logging
+from typing import Any, cast
+
+from homeassistant.components.cover import (
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_TOKEN,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, STATUSDICT_OPENSTATE, STATUSDICT_SERIALNO
+from .coordinator import ReisingerCoordinator
+from .entity import IntelliDriveEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+STATES_MAP = {0: STATE_CLOSED, 1: STATE_OPEN}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Intellidrive covers."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            SlidingDoorCoverEntity(
+                hass,
+                coordinator,
+                entry.data[CONF_HOST],
+                str(entry.data.get(CONF_TOKEN)),
+                cast(str, entry.unique_id),
+            )
+        ]
+    )
+
+
+class SlidingDoorCoverEntity(IntelliDriveEntity, CoverEntity):
+    """Wrapper class to adapt the intellidrive device into the Homeassistant platform."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        coordinator: ReisingerCoordinator,
+        host: str,
+        token: str,
+        device_id: str,
+    ) -> None:
+        """Initialize slidingdoor cover."""
+        self._state: str | None = None
+        self._state_before_move: str | None = None
+        self._host = host
+        self._token = token
+        self._device_api = coordinator.device
+        super().__init__(coordinator, device_id)
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the door async."""
+
+        if self._state in [STATE_CLOSED, STATE_CLOSING]:
+            return
+        self._state_before_move = self._state
+        self._state = STATE_CLOSING
```

Comment:
This will backfire if the call to `self._device_api.async_close()` fails. Instead of this, maybe ask the coordinator for an extra state refresh after calling the API? Or does the API only return open + closed, not open, closed, opening, closing?

Reply:
The API does only return an open-flag